python/desc/imsim/imSim.py

```diff
@@ -375,9 +382,13 @@ def sources_from_list(object_lines, obs_md, phot_params, file_name):
     assert len(x_pupil) == len(gs_object_arr)
     assert len(y_pupil) == len(gs_object_arr)
 
+    logger.debug('down-selecting by chip')
     out_obj_dict = {}
     for det in lsst_camera():
         chip_name = det.getName()
+        if target_chip is not None and chip_name != target_chip:
```

Comment:
If there is a specific sensor being considered (`target_chip`), then this if statement will skip the calculations below for the other 188 sensors.